src/OrbitBase/include/OrbitBase/Sort.h

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_SORT_H_
+#define ORBIT_BASE_SORT_H_
+
+#include <algorithm>
+#include <functional>
+#include <type_traits>
+#include <utility>
+
+namespace orbit_base_internal {
+
+template <typename Comparator, typename Projection>
+static auto MakeComparator(Comparator&& comparator, Projection&& projection) {
+  return [comparator = std::forward<Comparator>(comparator),
+          projection = std::forward<Projection>(projection)](const auto& a, const auto& b) {
+    return std::invoke(comparator, std::invoke(projection, a), std::invoke(projection, b));
+  };
+}
+
+}  // namespace orbit_base_internal
+
+namespace orbit_base {
+
+template <typename RandomIt, typename Projection, typename Comparator = std::less<>,
```

Comment:
Both functions should get a comment explaining what's the difference to std::sort